ui/src/loading.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import Loading from "./loading";
+
+test("loading", async () => {
+  const reload = jest.fn();
+  const props = { isPending: true, reload: reload };
+  const loading = () => <Loading status={props}>loaded</Loading>;
+
+  const { rerender } = render(loading());
+  await waitFor(() =>
+    expect(screen.getByRole("progressbar")).toBeInTheDocument()
+  );
+
+  expect(screen.queryByText("Reload")).not.toBeInTheDocument();
+
+  props.isPending = false;
+  props.error = new Error("test-error");
+  rerender(loading());
```

Comment:
instead of storing/updating props, how about just remaking the whole thing since its really a new test?  somethng like:
rerender(<Loading status={{isPending: false, error: new Error()}}/>);

is there a need to store the component in a variable?

Reply:
Per our discussion this morning, looks like inlining the JSX is just as effective and seems cleaner in the end.